Unable to configure TLS on NettyDockerCmdExecFactory without Jersey

I was hoping that I'd be able to cut Jersey out of my dependency tree and use Netty via the `NettyDockerCmdExecFactory`.

With TLS configured I hit the following issue. Do we really need Jersey to configure SSL, even when Netty is used as the client library?

```
Caused by: java.lang.NoClassDefFoundError: org/glassfish/jersey/SslConfigurator
	at com.github.dockerjava.core.LocalDirectorySSLConfig.getSSLContext(LocalDirectorySSLConfig.java:56)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory$InetSocketInitializer.initSsl(NettyDockerCmdExecFactory.java:340)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory$InetSocketInitializer.connect(NettyDockerCmdExecFactory.java:322)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory.connect(NettyDockerCmdExecFactory.java:220)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory.connect(NettyDockerCmdExecFactory.java:213)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory.access$000(NettyDockerCmdExecFactory.java:157)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory$1.getChannel(NettyDockerCmdExecFactory.java:183)
	at com.github.dockerjava.netty.InvocationBuilder.getChannel(InvocationBuilder.java:213)
	at com.github.dockerjava.netty.InvocationBuilder.get(InvocationBuilder.java:196)
	at com.github.dockerjava.netty.InvocationBuilder.get(InvocationBuilder.java:187)
	at com.github.dockerjava.netty.exec.ListImagesCmdExec.execute(ListImagesCmdExec.java:41)
	at com.github.dockerjava.netty.exec.ListImagesCmdExec.execute(ListImagesCmdExec.java:16)
	at com.github.dockerjava.netty.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
```

--- a/src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java
+++ b/src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java
@@ -6,12 +6,15 @@ import java.io.File;
 import java.io.Serializable;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
 import java.security.Security;
 
+import javax.net.ssl.KeyManagerFactory;
 import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
 
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
-import org.glassfish.jersey.SslConfigurator;
 
 import com.github.dockerjava.api.exception.DockerClientException;
 import com.github.dockerjava.core.util.CertificateUtils;
@@ -53,13 +56,20 @@ public class LocalDirectorySSLConfig implements SSLConfig, Serializable {
                 String certpem = new String(Files.readAllBytes(Paths.get(certPemPath)));
                 String capem = new String(Files.readAllBytes(Paths.get(caPemPath)));
 
-                SslConfigurator sslConfig = SslConfigurator.newInstance(true);
-                sslConfig.securityProtocol("TLSv1.2");
-                sslConfig.keyStore(CertificateUtils.createKeyStore(keypem, certpem));
-                sslConfig.keyStorePassword("docker");
-                sslConfig.trustStore(CertificateUtils.createTrustStore(capem));
+                String kmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.keyManagerFactory.algorithm",
+                    KeyManagerFactory.getDefaultAlgorithm()));
+                KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(kmfAlgorithm);
+                keyManagerFactory.init(CertificateUtils.createKeyStore(keypem, certpem), "docker".toCharArray());
 
-                return sslConfig.createSSLContext();
+                String tmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.trustManagerFactory.algorithm",
+                    TrustManagerFactory.getDefaultAlgorithm()));
+                TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(tmfAlgorithm);
+                trustManagerFactory.init(CertificateUtils.createTrustStore(capem));
+
+                SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
+                sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+
+                return sslContext;
 
             } catch (Exception e) {
                 throw new DockerClientException(e.getMessage(), e);
@@ -69,6 +79,15 @@ public class LocalDirectorySSLConfig implements SSLConfig, Serializable {
 
         return null;
 
+    }
+
+    private PrivilegedAction<String> getSystemProperty(final String name, final String def) {
+        return new PrivilegedAction<String>() {
+            @Override
+            public String run() {
+                return System.getProperty(name, def);
+            }
+        };
     }
 
     @Override

